piecedirectory/doctor.go

```diff
@@ -119,7 +163,7 @@ func (d *Doctor) checkPiece(ctx context.Context, pieceCid cid.Cid) error {
 		}
 	}
 
-	if !hasUnsealedDeal {
+	if !hasUnsealedDeal && !lacksActiveSector {
```

Comment:
Arguably this doesn't belong here, but the fact that we don't handle sector lifecycle properly, I'd rather we have this check somewhere than a ton of false-positives with respect to "unindexed" pieces.